### DISCUSSION TITLE
cel and flatbuffers

### DISCUSSION THREAD

**Discussion by yanivagman** [google/cel-go#587] [Q&A]
We are considering using cel-go in our project to evaluate performance critical expressions.
In addition, we are also thinking about using flatbuffers for serialization, which seems to have higher performance than protobuf.
In our use case, we need to serialize data (using protobuf/flatbuffers) before passing it to another process, then the second process needs to evaluate expressions using the received data.

I see that cel is built on top of protobuf and I wanted to ask if there are plans to add support for flatbuffers as well? I ask because the conversion from flatbuffers to protobuf may not be worth using flatbuffers. Does this even make sense?

**Comment by TristonianJones** [comment]
Hi @yanivagman, 

There aren't any plans to add support for flatbuffers in cel-go. Flatbuffer support was contributed to the cel-cpp by a collaborator, so if you're interested in doing something similar it's an option. You'd effectively need to create a `ref.TypeRegistry` (or possibly just a `ref.TypeProvider`) which understands how to handle reflection over flatbuffers similar to what you see in the `common/types/objects.go` and `common/types/pb/*.go` files.

Does that answer your question? For prototyping and conformance testing, I'd recommend having the default proto-based type provider run along side a custom type provider that supports flat buffers just to make sure you're capturing the behaviors of each properly.

-Tristan

**Comment by yanivagman** [comment]
Thanks @TristonianJones,
If you don't have any plans to do so, I guess you think it doesn't worth the effort and protobuf is good enough?

**Reply by TristonianJones** [reply]
Sorry for the delayed reply @yanivagman. For some reason I don't get notifications for discussion threads. 

The issue is more one of priority than of value. Flat buffers are supported in cel-cpp because someone felt it was valuable and a high priority, so they took the time to write the extensions needed. I was very happy to accept the contribution. At the moment, there are many other more urgent tasks that need to be completed before we could consider Flat buffer support in cel-go.
